File: blobs/howto/Java/blob-devguide/blob-devguide-containers/src/main/java/com/blobs/devguide/containers/ContainerCreate.java
package com.blobs.devguide.containers;

import com.azure.storage.blob.*;

public class ContainerCreate {
    // <Snippet_CreateContainer>
    public BlobContainerClient createContainer(BlobServiceClient blobServiceClient, String containerName) {
        // Create the container using the service client object
        BlobContainerClient blobContainerClient = blobServiceClient.createBlobContainer(containerName);

        return blobContainerClient;
    }
    // </Snippet_CreateContainer>

    // <Snippet_CreateRootContainer>
    public void createRootContainer(BlobServiceClient blobServiceClient) {
        // Creates a new BlobContainerClient object by appending the containerName to
        // the end of the URI
        BlobContainerClient blobContainerClient = blobServiceClient.getBlobContainerClient("$root");

        // If the container does not already exist, create it using the container client
        blobContainerClient.createIfNotExists();
    }
    // </Snippet_CreateRootContainer>
}

File: blobs/howto/TypeScript/NodeJS-v12/dev-guide/src/container-create.ts
// create-container.js
import {
  BlobServiceClient,
  ContainerClient,
  ContainerCreateOptions,
  ContainerCreateResponse
} from '@azure/storage-blob';
import * as dotenv from 'dotenv';
dotenv.config();

// Get BlobServiceClient
import { getBlobServiceClientFromDefaultAzureCredential } from './auth-get-client';
const blobServiceClient: BlobServiceClient =
  getBlobServiceClientFromDefaultAzureCredential();

//<snippet_create_container>
async function createContainer(
  blobServiceClient: BlobServiceClient,
  containerName: string
): Promise<ContainerClient> {
  const {
    containerClient,
    containerCreateResponse
  }: {
    containerClient: ContainerClient;
    containerCreateResponse: ContainerCreateResponse;
  } = await blobServiceClient.createContainer(containerName);

  if (containerCreateResponse.errorCode)
    throw Error(containerCreateResponse.errorCode);

  return containerClient;
}
//</snippet_create_container>

async function main(blobServiceClient): Promise<void> {
  // create container
  const containerName = 'sample-container';

  // create containers
  const containerClient = await createContainer(
    blobServiceClient,
    containerName
  );

  // Only one $root container per storage account
  const containerRootName = '$root';

  // Create root container
  await createContainer(blobServiceClient, containerRootName);
}
main(blobServiceClient)
  .then(() => console.log('success'))
  .catch((err: unknown) => {
    if (err instanceof Error) {
      console.log(err.message);
    }
  });
